sklearn/svm/_base.py

```diff
@@ -244,8 +239,21 @@ def fit(self, X, y, sample_weight=None):
                     "When 'gamma' is a string, it should be either 'scale' or "
                     "'auto'. Got '{}' instead.".format(self.gamma)
                 )
-        else:
+        elif isinstance(self.gamma, (numbers.Real, numbers.Integral)):
```

Comment:
I see the comment from @ogrisel at https://github.com/scikit-learn/scikit-learn/pull/21436#discussion_r735091353, but I think we do not need to include `numbers.Integral` here since `numbers.Real` covers integers:

```python
import numbers

isinstance(1, numbers.Real)
# True
```

Reply:
That's true :)

Reply:
Sorry...

Reply:
```suggestion
        elif isinstance(self.gamma, numbers.Real):
```